### PR TITLE
TINY-6745: Fixed an issue where the writer settings weren't being used for RTC

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6745

Description of Changes:
The RTC HTML serializer wasn't using the relevant writer editor settings, which meant indent, entities, etc... weren't working when serialising the content.

Note: I tested this works locally by pulling in the RTC plugin from cloud as an external plugin.

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] ~Branch prefixed with `feature/` for new features (if applicable)~
* [x] ~License headers added on new files (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by TheSpyder — APPROVED** [review]
At some point we'll have to figure out how to test this stuff, until then yeah manual testing seems fine.

**Review by spocke — APPROVED** [review]
Thanks for looking it to this. Been meaning to log and fix this but never got around to it.

**Comment by spocke** [comment]
We have tests on the rtc side that we will update to match the new content output. We need to bump the rtc tinymce build etc for that. I can fix all that once this is merged.

**Comment by lnewson** [comment]
> We have tests on the rtc side that we will update to match the new content output. We need to bump the rtc tinymce build etc for that. I can fix all that once this is merged.

Cool. I'll merge this Monday, unless it's really needed. I just wanted to make sure someone from the editor team reviewed it as well.

**Review by ltrouton — APPROVED** [review]

**Review by ashfordneil — APPROVED** [review]

**Pull request merged**
